### ISSUE TITLE
Style Organization

### ISSUE THREAD

**Issue by ameech** [dolfelt/scheduler-js#4]
What do you guys think about organizing our styles like this: http://sass-guidelin.es/#architecture, specifically this directory/file structure: http://sass-guidelin.es/#the-7-1-pattern

Also, what are you thoughts on using [BEM](https://css-tricks.com/bem-101/) for our style naming convention?

I've used these in the past and have found them quite scalable and easy enough to understand. They take a bit of discipline, but I think that's a fair trade off for less duplication and having any developer able to understand how the styles work.

**Note**: _This may be out of the scope for this repository, not sure how deeply we want to dive into styling._

@curtis1000 @dolfelt @shadowhand

**Comment by shadowhand** [comment]
Are we going to use [Bootstrap 4](http://blog.getbootstrap.com/2015/08/19/bootstrap-4-alpha/)? I think we should!

**Comment by dolfelt** [comment]
@shadowhand I don't think we should use Bootstrap in it's entirety. Right now I'm leaning towards only their grid styles, and perhaps their button/form styles. I'd like to keep it to a minimum since the style will always be custom.

**Comment by dolfelt** [comment]
Although looking at BS4, I'm impressed. Sass based, flex-box layout on the fly if we want it. Very nice! Even still, I'd want to pick and choose what works for us, and throw out (or stop using) what we find we are overriding most of.

**Comment by ameech** [comment]
Going back to the your previous comment @dolfelt, in what way does Sass get us part way there? By being able to nest?

I like the idea of using something like BEM as it gives us a document we can reference when new devs are onboarding as well as something we can hold each other accountable too. Also makes us think about how we are implementing styles instead of just tossing them in a file.

**Comment by dolfelt** [comment]
Yeah, I think nesting cleans things up really well. Also, for understanding of styles, there's always sourcemaps. Speaking of, we should add those to the gulp file.

**Comment by ameech** [comment]
We use nesting now and it's not so clean. Sourcemaps are great, but sometimes it's nice when things are obvious and don't need tools to understand what's going on.

**Comment by shadowhand** [comment]
@dolfelt right, if we use BS mixins we can pick and choose what parts we want to use and which to skip.
